nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -1107,6 +1107,10 @@ Make sure to also check the many updates in the [Nixpkgs library](#sec-release-2
   includes options for the built-in rate limiter, bot protection and
   automatically configuring a local Redis server.
 
+ `security.pam.enableSSHAgentAuth` was renamed to `security.pam.sshAgentAuth.enable` and an `authorizedKeysFiles`
+  option was added, to control which `authorized_keys` files are trusted.  It defaults to the previous behaviour,
+  **which is insecure**: see [#31611](https://github.com/NixOS/nixpkgs/issues/31611).
```

Comment:
Deferring to @figsoda @RaitoBezarius.

Reply:
- The rename is unnecessary and not security-adjacent.

I'm not convinced to another round of warning fatigue for everyone regarding this problem, I'm not sure most users can meaningfully mitigate this.

I would be open to add this **without** the warning.

Reply:
@nbraud We never emit new warnings really after a release has been cut, after a release is cut, and we ask people to move to it, especially after EOL of the previous version, people have a bunch of things to fix. This process shall happen only once to stable users in the lifetime of a release.

The security issue you are mentioning here does seem important to solve, but, I fear it would create **warning fatigue** given that 23.05 → 23.11 was quite a thing. This warning should have been emitted much earlier, we are too late in the release process now.

I fear also that you are forgetting that most of our users will not be able to understand the warning you are emitting them and choosing what you call a 'safe config' cause a loss of functionality and a mismatch in expectations regarding user-set `~/.ssh/authorized_keys` not tracked by Nix.

It is unfortunate, but I don't think we can do better than bringing a possibility for people aware of it and concerned about it to fix it in 23.11, but I don't think we should emit a warning that people can potentially want to ignore. We will already measure the interest for it in unstable, and I already know that it will annoy me because I know that I am using something that you deem insecure, and I would like to keep using it on some of my systems which have different threat models.

If I have this opinion that the security contents can be discussed more, I think this is a controversial change and cannot fly like this without proper process, i.e. **unstable** first and **time**.

Reply:
> I'd be quite happy for people to suggest better wording there, and I'll look for something as well.

I fear that I am the same as you, I am not a good person to suggest better wording or my contribution will be just slightly improving the status quo.

> Yes, that's exactly why changing the default behaviour was a separate PR that won't be backported.

Yep but people doesn't know that and when they receive a warning, they will attempt to fix it and may end up breaking their setup in difficult to recover ways (servers).